docs/dc/custom_factories/request.py

```diff
@@ -0,0 +1,17 @@
+# request.py
+from .models import Greeter, Request
+
+
+def process_request(registry, name):
+    # Each request gets its own container
+    container = registry.create_container()
+
+    # Manually make a Request and put in the container
+    request = Request(url=name)
+    container.set(request, Request)
```

Comment:
this should use `register_singleton` after the changes in wired 0.2

Reply:
Hmm, but i's value changes from request to request. I thought the container would be a good place to put such state.

Reply:
`container.register_singleton`, not `registry.register_singleton`. See the changelog in wired 0.2 about it. The `set` api only applies to a specific context in the container.